src/block/block_ckpt.c

```diff
@@ -83,11 +84,19 @@ __wt_block_checkpoint_load(WT_SESSION_IMPL *session, WT_BLOCK *block,
 		WT_ERR(__wt_block_ckpt_init(session, ci, "checkpoint"));
 	} else {
 		/*
-		 * We depend on the btree level for locking: things will go
-		 * bad fast should we open the live system in two handles, or
-		 * if we create, salvage, truncate or verify the live/running
-		 * file, for that matter.
+		 * We depend on the btree level for locking: things will go bad
+		 * fast if we open the live system in two handles, or salvage,
+		 * truncate or verify the live/running file.
 		 */
+		__wt_spin_lock(session, &block->live_lock);
+		is_open = block->live_open;
+		if (!is_open)
+			block->live_open = true;
+		__wt_spin_unlock(session, &block->live_lock);
+		if (is_open)
+			WT_ERR_MSG(session, WT_ERROR,
+			    "%s: open failed, live system already open",
+			    block->name);
```

Comment:
I have two comments. 1. Should we put all of this change under HAVE_DIAGNOSTICS?  2.  Given the comment, why not just `WT_ASSERT(session, block->live_open == false)` inside the spin lock instead of returning an error message?